pool/entities.go

```diff
@@ -0,0 +1,256 @@
+package pool
+
+type ID string
+
+type Name string
+
+// I don't know i have to bring Peer Available storage and other metric inside Peer
+type Peer string
+
+type ReplicationFactor int8
+
+// Requirement of joining the pool
+type Requirement struct {
+	Latency   string `json:"latency"`
+	Bandwidth string `json:"bandwidth"`
+}
+
+// ResourceRequest object (it is much more complicated for simplicity we think as file).
+type ResourceRequest struct {
+	ID  ID `json:"id"`
+	FileName  Name `json:"filename"`
+	Owner     Peer  `json:"owner"`
+	Size      string `json:"size"`
+	ReplicationFactor ReplicationFactor `json:"rp"`
+}
+
+type Resource struct{
+	ResourceRequest
+	Allocation []Peer 
+}
+
+
+// Event is a domain event marker.
+type Event interface {
+	isEvent()
```

Comment:
What is the purpose of this function?

Reply:
I hate to have empty interface and I don't have that much experience so I just copied it from [here](https://victoramartinez.com/posts/event-sourcing-in-go/)

Reply:
And if i remove the isEvent the code in the On method wont work.